### PR TITLE
See also & docstrings for `foldl`, `accumulate`

### PR DESCRIPTION
**Pull request by mcabbott**
The original goal was to make this link, since someone asked yesterday:
```
julia> foldl(=>, 1:4; init=0)
(((0 => 1) => 2) => 3) => 4

julia> accumulate(=>, (1,2,3,4))
(1, 1 => 2, (1 => 2) => 3, ((1 => 2) => 3) => 4)
```
Perhaps I got carried away trying to improve the doc examples for accumulate. They should be a little shorter than before, and explain more cases, e.g. that `init` is the first argument. Could probably be made shorter. 
```
julia> b = Iterators.accumulate(/, (2, 5, 2, 5); init = 100);

julia> collect(b)
4-element Vector{Float64}:
 50.0
 10.0
  5.0
  1.0
```

### CODE REVIEW

**Review by dkarrasch — COMMENTED** [review]
I like the new examples.

File: base/accumulate.jl [hunk 1]
```diff
@@ -116,35 +116,29 @@ end
 """
     cumsum(itr)
 
-Cumulative sum an iterator. See also [`cumsum!`](@ref)
-to use a preallocated output array, both for performance and to control the precision of the
-output (e.g. to avoid overflow).
+Cumulative sum of an iterator.
+
+See also [`accumulate`](@ref) to apply functions other than `+`
```
Comment (vtjnash) [inline base/accumulate.jl:121]:
```suggestion
See also [`accumulate`](@ref) to apply functions other than `+`.
```

**Review by vtjnash — APPROVED** [review]
LGTM, but why use Int8?

**Comment by mcabbott** [comment]
Probably too obscure there; the `cumsum` docstring has a comment that it promotes to avoid this:
```
julia> accumulate(+, Int8[1,12,123])
3-element Vector{Int8}:
    1
   13
 -120

julia> cumsum(Int8[1,12,123])
3-element Vector{Int64}:
   1
  13
 136
```

File: base/accumulate.jl [hunk 2]
```diff
@@ -243,49 +228,45 @@ cumprod(itr) = accumulate(mul_prod, itr)
...
 
 !!! compat "Julia 1.5"
     `accumulate` on a non-array iterator requires at least Julia 1.5.
 
 # Examples
 ```jldoctest
-julia> accumulate(+, [1,2,3])
-3-element Vector{Int64}:
+julia> accumulate(+, Int8[1,2,3])
+3-element Vector{Int8}:
```
Comment (mcabbott) [inline base/accumulate.jl:243]: Perhaps revert that:
```suggestion
julia> accumulate(+, [1,2,3])
3-element Vector{Int64}:
```

**Pull request merged**